src/redux/addCash.js

```diff
@@ -0,0 +1,66 @@
+import {
+  getPurchaseTransactions,
+  removePurchaseTransactions,
+  savePurchaseTransactions,
+} from '../handlers/localstorage/accountLocal';
+
+// -- Constants --------------------------------------- //
+const ADD_CASH_UPDATE_PURCHASE_TRANSACTIONS =
+  'addCash/ADD_CASH_UPDATE_PURCHASE_TRANSACTIONS';
+
+const ADD_CASH_CLEAR_STATE = 'addCash/ADD_CASH_CLEAR_STATE';
+
+// -- Actions ---------------------------------------- //
+export const addCashLoadState = () => async (dispatch, getState) => {
+  const { accountAddress, network } = getState().settings;
+  try {
+    const purchases = await getPurchaseTransactions(accountAddress, network);
+    dispatch({
+      payload: purchases,
+      type: ADD_CASH_UPDATE_PURCHASE_TRANSACTIONS,
+    });
+    // eslint-disable-next-line no-empty
+  } catch (error) {}
+};
+
+export const addCashClearState = () => (dispatch, getState) => {
+  const { accountAddress, network } = getState().settings;
+  removePurchaseTransactions(accountAddress, network);
```

Comment:
I don't think we should remove from storage since we're writing per account. 
(I am not doing it for savings)